core/src/main/java/org/infinispan/interceptors/distribution/BaseDistributionInterceptor.java

```diff
@@ -767,6 +784,9 @@ public MergingCompletableFuture(InvocationContext ctx, int participants, T[] res
 
       @Override
       protected Object result() {
+         if (hasUnsureResponse && lostData) {
```

Comment:
@danberindei this is weird; I can't find this comment on master. And I can't find ed7cf2b879d7b14e9a8f3179386ed50f9e274dcc even in my local history/branches.